### PR TITLE
pay another RDS API call to only consider instances in `AVAILABLE` state

### PR DESCRIPTION
**Pull request by gedl**
This is needed because when a cluster is deleting, it deletes the tags BEFORE it stops being reported as a cluster member.

This means if we tag a db instance with Fairlink-Exclude:true (and therefore fairlink excludes it from rotation), when we delete the instance there is the possibility of on a following discovery cycle, the instance not to be excluded anymore (because RDS removes the tags) and therefore be included again in the rotation. 

After this, the instance will eventually drop all its connections and cause failures in the application.

This PR pays another RDS API call per discovery cycle (effectively doubling the calls made to RDS API) to check, on top of tag exclusions, that all instances returned to the application are in the `AVAILABLE` state.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,20 +96,30 @@ dice-fairlink uses the [`RDS` API](https://docs.aws.amazon.com/AmazonRDS/latest/
    {
       "Effect":"Allow",
       "Action":[
-         "rds:DescribeDBClusters"
+         "rds:DescribeDBClusters",
```
Comment (julianhowarth) [inline README.md]: trailing comma

File: README.md [hunk 2]
```diff
@@ -134,7 +144,7 @@ It is possible to prevent members of an Aurora cluster to receive connections de
...
+dice-fairlink uses the [`Resource Group` API](https://docs.aws.amazon.com/ARG/latest/APIReference/Welcome.html) to obtain the list of all excluded members. Unfortunately this API does not offer a way to obtain only the information pertaining to a specific cluster, and as a consequence dice-fairlink will hold information about **all** RDS instances tagged with `Fairlink-Exclude=true` in the entire region of the AWS accounbt in question. It is also not possible to restrict this via an IAM policy. It is assumed the size of this collection will not be problematic, and that making a single call (instead of 1 call per database instance) outweighs this waste.
```
Comment (julianhowarth) [inline README.md]: sp `accounbt`

File: src/main/java/technology/dice/dicefairlink/discovery/members/awsapi/AwsApiReplicasFinder.java [hunk 3]
```diff
@@ -58,23 +67,48 @@ private DBCluster describeCluster(String clusterId) {
...
+      DescribeDbInstancesRequest request =
+          DescribeDbInstancesRequest.builder()
+              .filters(
+                  Filter.builder()
+                      .name(DB_CLUSTER_ID_FILTER)
+                      .values(cluster.dbClusterIdentifier())
+                      .build())
+              .build();
+      final Optional<DBClusterMember> writer =
+          cluster.dbClusterMembers().stream().filter(member -> member.isClusterWriter()).findAny();
```
Comment (julianhowarth) [inline src/main/java/technology/dice/dicefairlink/discovery/members/awsapi/AwsApiReplicasFinder.java]: use `Stream.anyMatch(...)`

File: src/main/java/technology/dice/dicefairlink/discovery/members/awsapi/AwsApiReplicasFinder.java [hunk 4]
```diff
@@ -58,23 +67,48 @@ private DBCluster describeCluster(String clusterId) {
...
+
+    } catch (Exception e) {
+      LOGGER.log(
+          Level.SEVERE, "Failed to list cluster replicas. Returning an empty set of replicas", e);
+      return EMPTY_SET;
+    }
   }
 
-  private static boolean isDbWriter(Optional<DBClusterMember> writer, DBClusterMember dbInstance) {
+  private static boolean isDbWriter(Optional<DBClusterMember> writer, DBInstance dbInstance) {
```
Comment (julianhowarth) [inline src/main/java/technology/dice/dicefairlink/discovery/members/awsapi/AwsApiReplicasFinder.java]: isn't this actually `isNotWriter`? (Or `isReader`)

File: src/main/java/technology/dice/dicefairlink/discovery/members/awsapi/AwsApiReplicasFinder.java (hunk 3 above)
Comment (gedl) [reply src/main/java/technology/dice/dicefairlink/discovery/members/awsapi/AwsApiReplicasFinder.java]: That will only give me a boolean, I need the actual instance id

**Pull request merged**